cmake/vcpkg/ports/wasm32-wasi/fmt/README.md

```diff
@@ -0,0 +1 @@
+This overlay port is a direct copy from https://github.com/microsoft/vcpkg/tree/master/ports/fmt with modifications to allow compilation on wasm32-wasi as a target by removing POSIX filesystem calls.
```

Comment:
According to fmt github repo it supports build with emscripten.
Can you share a compilation error?

Reply:
In short: `emscripten != wasi`. This PR allows compiling with WASI, not Emscripten.

WebAssembly has two popular ABI flavors: Emscripten and WASI.
Emscripten is more feature-forward at the cost of being non-standard and only supporting Javascript as a runtime host.

WASI is more limited in capabilities but is standardized and can be run in many more runtimes (Javascript, Golang, Rust, CLI, ...).

Here's the error without the overlay port:

```Change Dir: '/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/wasm32-wasi-dbg'

Run Build Command(s): /opt/homebrew/bin/ninja -v -v -j11 install
[1/4] /opt/homebrew/opt/llvm/bin/clang++ --target=wasm32-wasi --sysroot=/Users/riomartinez/.cache/wasi-sdk/21.0/wasi-sysroot  -I/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include -g -fvisibility=hidden -fvisibility-inlines-hidden -stdlib=libc++ -fno-exceptions -D_WASI_EMULATED_SIGNAL -MD -MT CMakeFiles/fmt.dir/src/os.cc.obj -MF CMakeFiles/fmt.dir/src/os.cc.obj.d -o CMakeFiles/fmt.dir/src/os.cc.obj -c /Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/os.cc
FAILED: CMakeFiles/fmt.dir/src/os.cc.obj 
/opt/homebrew/opt/llvm/bin/clang++ --target=wasm32-wasi --sysroot=/Users/riomartinez/.cache/wasi-sdk/21.0/wasi-sysroot  -I/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include -g -fvisibility=hidden -fvisibility-inlines-hidden -stdlib=libc++ -fno-exceptions -D_WASI_EMULATED_SIGNAL -MD -MT CMakeFiles/fmt.dir/src/os.cc.obj -MF CMakeFiles/fmt.dir/src/os.cc.obj.d -o CMakeFiles/fmt.dir/src/os.cc.obj -c /Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/os.cc
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/os.cc:273:31: error: no member named 'dup' in the global namespace
  273 |   int new_fd = FMT_POSIX_CALL(dup(fd));
      |                ~~~~~~~~~~~~~~~^~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:56:36: note: expanded from macro 'FMT_POSIX_CALL'
   56 | #    define FMT_POSIX_CALL(call) ::call
      |                                  ~~^
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/os.cc:282:36: error: no member named 'dup2' in the global namespace
  282 |   FMT_RETRY(result, FMT_POSIX_CALL(dup2(fd_, fd)));
      |                     ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:56:36: note: expanded from macro 'FMT_POSIX_CALL'
   56 | #    define FMT_POSIX_CALL(call) ::call
      |                                  ~~^
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:71:61: note: expanded from macro 'FMT_RETRY'
   71 | #define FMT_RETRY(result, expression) FMT_RETRY_VAL(result, expression, -1)
      |                                                             ^~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:65:19: note: expanded from macro 'FMT_RETRY_VAL'
   65 |       (result) = (expression);                            \
      |                   ^~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/os.cc:292:36: error: no member named 'dup2' in the global namespace
  292 |   FMT_RETRY(result, FMT_POSIX_CALL(dup2(fd_, fd)));
      |                     ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:56:36: note: expanded from macro 'FMT_POSIX_CALL'
   56 | #    define FMT_POSIX_CALL(call) ::call
      |                                  ~~^
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:71:61: note: expanded from macro 'FMT_RETRY'
   71 | #define FMT_RETRY(result, expression) FMT_RETRY_VAL(result, expression, -1)
      |                                                             ^~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:65:19: note: expanded from macro 'FMT_RETRY_VAL'
   65 |       (result) = (expression);                            \
      |                   ^~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/os.cc:309:31: error: no member named 'pipe' in the global namespace
  309 |   int result = FMT_POSIX_CALL(pipe(fds));
      |                ~~~~~~~~~~~~~~~^~~~~~~~~~
/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include/fmt/os.h:56:36: note: expanded from macro 'FMT_POSIX_CALL'
   56 | #    define FMT_POSIX_CALL(call) ::call
      |                                  ~~^
4 errors generated.
[2/4] /opt/homebrew/opt/llvm/bin/clang++ --target=wasm32-wasi --sysroot=/Users/riomartinez/.cache/wasi-sdk/21.0/wasi-sysroot  -I/Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/include -g -fvisibility=hidden -fvisibility-inlines-hidden -stdlib=libc++ -fno-exceptions -D_WASI_EMULATED_SIGNAL -MD -MT CMakeFiles/fmt.dir/src/format.cc.obj -MF CMakeFiles/fmt.dir/src/format.cc.obj.d -o CMakeFiles/fmt.dir/src/format.cc.obj -c /Users/riomartinez/.cache/vcpkg/projects/crdt/buildtrees/fmt/src/10.0.0-cbbdb219f9.clean/src/format.cc
ninja: build stopped: subcommand failed.

```

Reply:
For reference, here is the code from `wasi-libc` which can be seen as a source-of-truth for what system calls are supported in WASI: https://github.com/WebAssembly/wasi-libc/blob/6593687e25f07526c4b92a20fe5ddf507599d5b3/libc-top-half/musl/include/unistd.h#L54

As can be seen from the link, `dup`, `dup2` and `pipe` are not supported by WASI.